### PR TITLE
Auth module

### PR DESCRIPTION
**Pull request by pranay-mm**
#### PR details:

* Design login screen with required components 
* Setup `flutter_riverpod` with `riverpod_generator` to start adding auth screen-related providers 
  * Using `StateNotifier` for login form validation and error
  * Using `StateNotifier` along with `StateNotifierProvider` for user authentication with the different state(e.g Firebase login state with error showing)

#### Screenshots:
| Auth screen | 
| ------------- | 
| <video src='https://github.com/pranay-mm/rick_morty_futter/assets/110823088/415afa29-0491-4043-b6d9-c66402497fbd' width=180/> |

### CODE REVIEW

File: lib/features/onboarding/onboarding_screen.dart [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        body: SafeArea(
+      child: SingleChildScrollView(
+        child: Column(
+          children: [
+            SizedBox(
+              height: context.height * 0.85,
```
Comment (aditya-bhawsar-mm) [inline lib/features/onboarding/onboarding_screen.dart:61]: https://api.flutter.dev/flutter/widgets/FractionallySizedBox-class.html or flexible

File: lib/features/onboarding/onboarding_screen.dart [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                      ),
+                      const Spacer(),
+                      Align(
+                        alignment: Alignment.centerRight,
+                        child: Stack(alignment: Alignment.center, children: [
+                          FloatingActionButton(
+                            onPressed: () {
+                              log('on click');
+                              navigateOnLogin(context);
+                            },
```
Comment (aditya-bhawsar-mm) [inline lib/features/onboarding/onboarding_screen.dart:151]: Align is useless here

File: lib/features/onboarding/onboarding_screen.dart [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                        child: TextButton(
+                          onPressed: () {
+                            navigateOnLogin(context);
+                          },
+                          child: Text('Skip',
+                              style: context.textTheme.titleMedium
+                                  ?.copyWith(fontWeight: FontWeight.w400)),
+                        ),
+                      ),
+                      const Spacer(),
```
Comment (aditya-bhawsar-mm) [inline lib/features/onboarding/onboarding_screen.dart:143]: Spacer is not useful when there are only 2 widgets

**Review by aditya-bhawsar-mm — APPROVED** [review]

**Comment by pranay-mm** [comment]
@aditya-bhawsar-mm All PR comment has been addressed and commit code in current branch. Thank you for the review.

**Pull request merged**